torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -286,16 +299,16 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                 raise NotImplementedError(
                     f"FSDP only supports 1D TP, not {self._tp_spec.placements}"
                 )
-            split_factor = self._tp_spec.num_shards_map[0]
+            split_factor = self._tp_spec.num_shards_map[shard_dim]
             assert (
                 2 <= self._spmd_mesh.ndim <= 3
             ), f"_spmd_mesh.ndim can only be 2 or 3 but got {self._spmd_mesh.ndim}."
             self._spmd_placements: Tuple[Placement, ...]
             dp_shard_tp_placement = (
                 (
-                    _StridedShard(0, split_factor=split_factor)
+                    _StridedShard(shard_dim, split_factor=split_factor)
```

Comment:
I never truly understand strided sharding. I guess it's meant for `model.parameters()` outside of fwd/bwd ? mostly for DTensor ops outside of FSDP, like state dict?

Reply:
yep